### PR TITLE
fix: handle broken listAssets and listVotes functions

### PR DESCRIPTION
**Pull request by domw30**
[Associated Notion Card](https://wilderworld.atlassian.net/jira/software/c/projects/ZAP/boards/34?modal=detail&selectedIssue=ZAP-142)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
- Bugfix

## 3. What is the old behaviour?
List assets was throwing errors causing loading to hang for totals and assets table on DAO dApp.

## 4. What is the new behaviour?
Replaced list all assets with list coins.
Refactored some UI to reflect the changes.
Loom - https://www.loom.com/share/12d3df6074384719b48a1dd164aa08b2

### CODE REVIEW

File: src/containers/DAO/pages/DAOList/DAOTable/DAOTableRow.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const DAOTableRow: React.FC<DAOTableRowProps> = ({ data }) => {
 		if (totalUsd) {
 			add({ zna: data.zna, total: totalUsd });
 		}
-	}, [totalUsd]);
+	}, [add, data.zna, totalUsd]);
```
Comment (domw30) [inline src/containers/DAO/pages/DAOList/DAOTable/DAOTableRow.tsx:55]: these missing dependents were causing the stat widget total to be incorrect on DAO homepage

File: src/containers/DAO/pages/DAOList/DAOTable/DAOTableCard.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ const DAOTableCard: React.FC<DAOTableCardProps> = ({ data }) => {
 		if (totalUsd) {
 			add({ zna: zna, total: totalUsd });
 		}
-	}, [totalUsd]);
+	}, [add, data.zna, totalUsd]);
```
Comment (domw30) [inline src/containers/DAO/pages/DAOList/DAOTable/DAOTableCard.tsx:44]: these missing dependents were causing the stat widget total to be incorrect on DAO homepage

File: src/containers/DAO/pages/DAOPage/hooks/useAssets.ts [hunk 3]
```diff
@@ -45,16 +45,17 @@ const useAssets = (dao?: zDAO): UseAssetsReturn => {
 			setAssets(undefined);
 			setIsLoading(true);
 			dao
-				?.listAssets()
+				?.listAssetsCoins()
+				// ?.listAssets()
```
Comment (domw30) [inline src/containers/DAO/pages/DAOPage/hooks/useAssets.ts:49]: commented out code here as we'll likely re-use

File: src/containers/DAO/pages/DAOList/DAOTable/DAOTableCard.tsx (hunk 2 above)
Comment (colbr) [reply src/containers/DAO/pages/DAOList/DAOTable/DAOTableCard.tsx:44]: Nice find! Can't wait to deprecate this 🥲

File: src/containers/DAO/pages/DAOList/TotalProvider.tsx [hunk 4]
```diff
@@ -16,16 +18,20 @@ type TotalsProviderProps = {
 
 export const TotalsProvider = ({ children }: TotalsProviderProps) => {
 	const [totals, setTotals] = useState<Total[]>([]);
+	const [isLoading, setIsLoading] = useState<boolean>(true);
 
 	const add = (total: Total) => {
 		if (!totals.map((t) => t.zna).includes(total.zna)) {
 			setTotals([...totals, total]);
+			setIsLoading(false);
```
Comment (colbr) [inline src/containers/DAO/pages/DAOList/TotalProvider.tsx:26]: nit: can get rid of this `setIsLoading`, as the second one will run regardless of whether or not the `if` condition is met

**Review by colbr — APPROVED** [review]
Nice!

File: src/containers/DAO/pages/DAOPage/Assets/AssetsTable/AssetsTable.tsx [hunk 5]
```diff
@@ -29,29 +32,48 @@ const HEADERS = [
 	},
 ];
 
-const AssetsTable: React.FC<AssetsTableProps> = ({ assets, isLoading }) => {
+const AssetsTable: React.FC<AssetsTableProps> = ({
+	assets,
+	safeAddress,
```
[the remark this answers is not in the document]
Comment (domw30) [reply src/containers/DAO/pages/DAOPage/Assets/AssetsTable/AssetsTable.tsx:37]: Yeah `safeAddress` is being passed down as to be used as part of the url to navigate to etherscan to view the DAO collections :)

File: src/containers/DAO/pages/DAOList/TotalProvider.tsx [hunk 6]
```diff
@@ -3,11 +3,13 @@ import React, { useState } from 'react';
 type Total = {
 	zna: string;
 	total: number;
+	isLoading?: boolean;
 };
 
 export const TotalsContext = React.createContext({
 	totals: [] as Total[],
 	add: (total: Total) => {},
+	isLoading: true as boolean | undefined,
```
[the remark this answers is not in the document]
Comment (domw30) [reply src/containers/DAO/pages/DAOList/TotalProvider.tsx:12]: Thanks, I'll remove :)

**Pull request merged**